<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>721</width>
    <height>438</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ArkPaintConverter</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QScrollArea" name="scrollArea_img1">
            <property name="toolTip">
             <string>original Image</string>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents_2">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>298</width>
               <height>177</height>
              </rect>
             </property>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QScrollArea" name="scrollArea_img2">
            <property name="toolTip">
             <string>converted image</string>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents_3">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>298</width>
               <height>177</height>
              </rect>
             </property>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QSpinBox" name="spinBox">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="font">
             <font>
              <pointsize>5</pointsize>
             </font>
            </property>
            <property name="text">
             <string>ViewScale%</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="font">
             <font>
              <pointsize>5</pointsize>
             </font>
            </property>
            <property name="text">
             <string>ViewScale%</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_2">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLineEdit" name="lineEditImgPath">
            <property name="toolTip">
             <string>File Path</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>select file path</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnOpen">
            <property name="text">
             <string>Open</string>
            </property>
            <property name="checkable">
             <bool>false</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>TargetSize:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>PixelX:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditPixelX">
            <property name="inputMask">
             <string/>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>PixelY:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditPixelY"/>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>KeepAspectRatio</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton">
            <property name="text">
             <string>ConvertImage</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnWriteFile">
            <property name="text">
             <string>WriteFile</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Dither</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditDither">
            <property name="toolTip">
             <string>Improves image's color depth(0.0 - 1.0)</string>
            </property>
            <property name="whatsThis">
             <string/>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_2">
            <property name="currentText">
             <string/>
            </property>
            <property name="currentIndex">
             <number>-1</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>83</width>
           <height>345</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>721</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionConvert_Ark_pnt_to_png"/>
    <addaction name="actionSave_as_png"/>
    <addaction name="actionRead_Color_Table"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Info</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionConvert_Ark_pnt_to_png">
   <property name="text">
    <string>Convert Ark.pnt to png</string>
   </property>
  </action>
  <action name="actionSave_as_png">
   <property name="text">
    <string>Save as png</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionRead_Color_Table">
   <property name="text">
    <string>Read Color Table</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>btnOpen</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OpenImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>337</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>ConvertImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>528</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>371</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>SelectImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>277</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEditPixelX</sender>
   <signal>textEdited(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>PixelXChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>180</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>86</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEditPixelY</sender>
   <signal>textEdited(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>PixelYChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>285</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnWriteFile</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>WriteFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>618</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>SelectImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>247</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>247</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionConvert_Ark_pnt_to_png</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>ConvertPntToPng()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_as_png</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>SaveAsPng()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>ActionAboutTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRead_Color_Table</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>ReadColorTable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>360</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OpenImage()</slot>
  <slot>ConvertImage()</slot>
  <slot>SelectImage()</slot>
  <slot>PixelXChanged()</slot>
  <slot>PixelYChanged()</slot>
  <slot>WriteFile()</slot>
  <slot>ConvertPntToPng()</slot>
  <slot>SaveAsPng()</slot>
  <slot>ActionAboutTriggered()</slot>
  <slot>ReadColorTable()</slot>
 </slots>
</ui>
